hazelcast/src/main/java/com/hazelcast/client/impl/protocol/MessageTaskFactoryImpl.java

```diff
@@ -22,21 +44,41 @@ public MessageTaskFactoryImpl(Node node) {
     }
 
     public void initFactories() {
-        tasks.put(ClientMessageType.AUTHENTICATION_DEFAULT.id(), new MessageTaskFactory() {
+        tasks.put(ClientMessageType.AUTHENTICATION_DEFAULT_REQUEST.id(), new MessageTaskFactory() {
             public PartitionSpecificRunnable create(ClientMessage clientMessage, Node node, Connection connection) {
                 return new AuthenticationMessageTask(clientMessage, node, connection);
             }
         });
-        tasks.put(ClientMessageType.AUTHENTICATION_CUSTOM.id(), new MessageTaskFactory() {
+        tasks.put(ClientMessageType.AUTHENTICATION_CUSTOM_REQUEST.id(), new MessageTaskFactory() {
             public PartitionSpecificRunnable create(ClientMessage clientMessage, Node node, Connection connection) {
                 return new AuthenticationCustomCredentialsMessageTask(clientMessage, node, connection);
             }
         });
-        tasks.put(MapMessageType.MAP_PUT.id(), new MessageTaskFactory() {
+        tasks.put(ClientMessageType.MAP_PUT_REQUEST.id(), new MessageTaskFactory() {
             public PartitionSpecificRunnable create(ClientMessage clientMessage, Node node, Connection connection) {
                 return new MapPutMessageTask(clientMessage, node, connection);
             }
         });
+        tasks.put(ClientMessageType.ADD_ENTRY_LISTENER_REQUEST.id(), new MessageTaskFactory() {
```

Comment:
Is there any interface/class named `MessageTask`? Otherwise `MessageTaskFactory` name doesn't name sense to me. It's creating a type that does not exist.